app/components/post_draft/post_input/index.ts

```diff
@@ -34,21 +33,16 @@ const enhanced = withObservables([], ({database, channelId, rootId}: WithDatabas
         switchMap((cfg) => of$(parseInt(cfg?.MaxNotificationsPerChannel || '0', 10))),
     );
 
-    let channel;
-    if (rootId) {
-        channel = observeChannel(database, channelId);
-    } else {
-        channel = observeCurrentChannel(database);
-    }
```

Comment:
No matter if we are or not on a thread. We want the channel that we are passed in the props. Also, that observable is slightly better (observe current channel has to observe the current channel id and then the channel by that id).